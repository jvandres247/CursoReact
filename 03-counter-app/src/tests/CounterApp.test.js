import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";
import "../setupTest";

describe("Pruebas en <CounterApp />", () => {
  let wrapper = shallow(<CounterApp />);;

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  })

  test("Debe mostrar <CounterApp /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el valor por defecto de 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);

    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("100");
  });

  test("Debe de incrementar con el boton de +1", () => {

    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe('2');
  });

  test("Debe de decrementar con el boton de -1", () => {

    wrapper.find('button').at(2).simulate('click');
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe('0');
  });

  test('Debe hacer un reset con el boton', () => {
    const wrapper = shallow(<CounterApp value={100} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    const counterText = wrapper.find("h2").text().trim();
    
    expect(counterText).toBe('100');
  })
  
});
