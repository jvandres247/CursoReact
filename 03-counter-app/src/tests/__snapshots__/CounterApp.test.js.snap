// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
